ms_active_directory/core/ad_session.py

```diff
@@ -617,6 +618,95 @@ def create_computer(self, computer_name: str, computer_location: str = None, com
         return ManagedADComputer(samaccount_name, self.domain, computer_location, computer_password, spns,
                                  encryption_types)
 
+    def _delete_object(self, object_dn: str, sanity_check_for_existence: bool = True):
+        if sanity_check_for_existence and not self.dn_exists_in_domain(object_dn):
+            raise ObjectDeletionException('An object does not exist within the domain with distinguished name {} - '
+                                          'it needs to exist to be deleted.'.format(object_dn))
+        res = self.ldap_connection.delete(object_dn)
+        # TODO: returning the actual response is probably a good idea so we can do something with it.
+        # doing more with it in case of error would be a good idea too
+        success, result, _, _ = ldap_utils.process_ldap3_conn_return_value(self.ldap_connection, res)
+        if success:
+            return success
+        # don't include attributes in the exception because a password could be there and it could get logged.
+        raise ObjectDeletionException('An exception was encountered deleting an object with distinguished name {}.'
+                                      'LDAP result: {}'.format(object_dn, result))
+
+    def delete_user(self, username: str, object_location: str = None, skip_validation: bool = False):
```

Comment:
can we rename this `delete_user_by_common_name` to match the same style as other functions? this specifically assumes the username is a common name, but we might want an option that takes the user object later